src/flexible_type/string_parser.hpp

```diff
@@ -229,10 +235,19 @@ struct string_parser
       if (!quote_char) boost::algorithm::trim_right(final_str);
       else if (quote_char) {
         // if was quoted field, we unescape the contents
-        turi::unescape_string(final_str, config.escape_char,
-                                  quote_char, config.double_quote);
+        turi::unescape_string(final_str, config.use_escape_char, 
+                              config.escape_char,
+                              quote_char, config.double_quote);
+      }
+      if (!config.na_val.empty() && config.na_val.count(final_str)) {
+        attr = turi::flexible_type(turi::flex_type_enum::UNDEFINED);
```

Comment:
Is there a possible reason to separate missing and float NA?  It seems that na_value, if specified, might not be interpreted as missing in a float column.

Reply:
It should work correctly in a float column actually. 
```
a,b
1.0,2.0
2.0,3.0
```
```
tc.SFrame.read_csv('z.csv',na_values=['2.0'])
Data:
+------+------+
|  a   |  b   |
+------+------+
| 1.0  | None |
| None | 3.0  |
+------+------+
```
This is actually handled in the CSV tokenization phase.

Where this particular bit is used is to support Nones in recursive parses.
```
a,b
{'a':1},{'b':null}
{'b':null},{'b':2}
tc.SFrame.read_csv('z.csv',na_values=['null'])
```

Indeed in this case, if you do na_values=['1'] that will probably not handle correctly. But handling this correctly in the recursive parse is problematic since that does not fit in the grammar nicely.

Reply:
My question is more about whether we should have a separate na_values field, with goes to NAN, and a missing_values field, which goes to None.  Having na_values go to None seems like it could confuse people.

Reply:
This was a design decision a long time ago that expands to all of SFrame. That NaN is a valid float value and should not be confused with None.

Reply:
Followup:  Did not realize that this behavior is unchanged from previous version; no reason to change.